core/src/main/java/gatt/GattConnection.kt

```diff
@@ -45,9 +45,12 @@ interface GattConnection {
     suspend fun disconnect(): Unit
 }
 
-class GattErrorStatusException internal constructor(
-    val event: OnConnectionStateChange
-) : IOException("Received $event")
+class GattStatusException(message: String?) : IOException(message) {
+    constructor(
+        status: GattStatus,
+        prefix: String
+    ) : this("$prefix failed with status ${status.asGattStatusString()}")
+}
```

Comment:
Should these constructors be `internal` like the other exceptions?

Reply:
Yes, but...
This exception is also thrown from `throw` module.

I've tried utilizing `friend-paths` (to give `throw` module access to `internal` definitions in `core`) but could not get it working. 😢 

Spent about 3 hours trying to get the friend paths working before giving up and just accepting this little bit of cruft on the public API.

Reply:
I really wish Kotlin had _way_ more visibility modifiers. Something like Android's `@RestrictTo` but at a language level.